meta-client/src/test/java/org/apache/helix/metaclient/impl/zk/TestStressZkClient.java

```diff
@@ -1,24 +1,394 @@
+      }
+
+      zkMetaClient.delete(zkParentKey);
+      zkMetaClient.create(zkParentKey, "parent_node");
+
+      // Set with path to non-existent node - should fail
+      for (int i = 0; i < TEST_ITERATION_COUNT; i++) {
+        try {
+          zkMetaClient.update("/a/b/c", new DataUpdater<Integer>() {
+            @Override
+            public Integer update(Integer currentData) {
+              return currentData + 1;
+            }
+          });
+
+          Assert.fail("Should have failed due to path to non-existent node");
+        } catch (Exception ignoredException) {
+        }
+      }
+
+      zkMetaClient.delete(zkParentKey);
+      zkMetaClient.create(zkParentKey, "parent_node");
+
+      // Set with invalid path - should fail
+      for (int i = 0; i < TEST_ITERATION_COUNT; i++) {
+        try {
+          zkMetaClient.update("/a/b/c", new DataUpdater<Integer>() {
+            @Override
+            public Integer update(Integer currentData) {
+              return currentData + 1;
+            }
+          });
+          Assert.fail("Should have failed due to invalid path - no leading /.");
+        } catch (Exception ignoredException) {
+        }
+      }
+    }
+    cleanUpTestNodes(zkParentKey);
+  }
+
+  @Test
+  public void testUpdateMultiNode() {
+    final String zkParentKey = "/stressZk_testUpdateSingleNode";
+    ZkMetaClientConfig config =
+        new ZkMetaClientConfig.ZkMetaClientConfigBuilder().setConnectionAddress(ZK_ADDR).build();
+    try (ZkMetaClient<Integer> zkMetaClient = new ZkMetaClient<>(config)) {
+      zkMetaClient.connect();
+      zkMetaClient.create(zkParentKey, "test_parent");
+
+      for (int i = 0; i < TEST_ITERATION_COUNT; i++) {
+        zkMetaClient.create(zkParentKey + "/" + i, 0);
+      }
+
+      for (int i = 0; i < TEST_ITERATION_COUNT; i++) {
+        String childKey = zkParentKey + "/" + i;
+        MetaClientInterface.Stat entryStat = zkMetaClient.exists(childKey);
+        Assert.assertEquals(entryStat.getVersion(), 0);
+
+        Integer newData = zkMetaClient.update(childKey, new DataUpdater<Integer>() {
+          @Override
+          public Integer update(Integer currentData) {
+            return currentData + 1;
+          }
+        });
+
+        Assert.assertEquals((int) newData, 1);
+        entryStat = zkMetaClient.exists(childKey);
+        Assert.assertEquals(entryStat.getVersion(), 1);
+      }
+    }
+    cleanUpTestNodes(zkParentKey);
+  }
+
+  @Test
+  public void testMultipleDataChangeListeners() throws Exception {
+    final String zkParentKey = "/stressZk_testMultipleDataChangeListeners";
+    final int listenerCount = 10;
+    final String testData = "test-data";
+    final AtomicBoolean dataExpected = new AtomicBoolean(true);
+
+    try (ZkMetaClient<String> zkMetaClient = createZkMetaClient()) {
+      zkMetaClient.connect();
+      zkMetaClient.create(zkParentKey, "parent_node");
+      Map<String, Set<DataChangeListener>> listeners = new HashMap<>();
```

Comment:
Do we need to store this?

Reply:
I've removed this portion in most recent commits